[IMP] website_sale, *: introduce "bento" design

Brieuc-brd opened the pull request:
*: website_sale_wishlist

This commit introduces a new Product Design Template called "Bento".

task-4924425





---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

stefanorigano reviewed (approved):
Hello @Brieuc-brd , thanks for this, cool 😎
I've pushed and squashed two little quirks. LGTM 👍 

Ready for tech

Brieuc-brd commented:
Hey @livaios 👋 

Thanks for your review, I’ve reworked the code 💪 what do you think ? 🙂 .

> 1. When you add rating to be viewed there's extra space between attributes and product title

It's now fixed ✅ 

> 2. Not sure if its a limitation or not but the attribute on hover is not working, its always shown

Good catch 👍  I've fixed that, now the attributes are shown according to the correct configuration.

> 3. This is not 1/1, after testing when its saved its set correctly but its the only display where its not applied on the editor
Same thing for snippets when ratio is changed the size of the container remains the same doesn't


It seems there is a bug since `19.0` that affects all product designs.
For some reason, once you save the shop page with a specific ratio other than 1/1, the `o_wsale_products_opt_thumb_*` class is applied to the `<body>` instead of the `.o_wsale_products_grid_table` section.

<img width="605" height="244" alt="Capture d’écran 2025-09-26 à 15 49 03" src="https://github.com/user-attachments/assets/4203a741-e19e-4c34-8b54-c204e53b307c" />

When the ratio is set to 1/1, any `o_wsale_products_opt_thumb_` classes should be removed so that the default ratio applies. However, since the class is being defined on the `<body>`, it seems to be removed only after saving.

The main issue is that this behavior creates conflicts with the Catalog snippet and the Wishlist table, as they inherit the class defined on the `<body>`. 🤔 

Since this issue already exists in 19.0, it should be addressed with a FIX targeting that version.

@stefanorigano 👋  what do you think about that ☝️ ?

> 4. When description is enabled, since its activated on hover not all hover effects are activated correctly its a bit buggy

I’ve reviewed the behavior of the hover effects for this design, it should now work correctly when hovering the cards. 👍 

> 5. Bento for wishlist needs revisiting

Done ✅ 

> 6. For snippets, when description, rating, add to cart are enabled, or when description is too long

Indeed, not only for snippets: it will be up to the user to choose the appropriate aspect ratio depending on the content he wants to display.
I’ve limited the number of description lines based on the card size to avoid as many layout conflicts as possible.
However, if the user decides to use a 5-column layout, a large gap, and display all information, content overflow should be expected.
Side note: as the description is editable in edit mode, it's only truncated in non-edit mode. 🙂 

Thanks in advance ! 🙏
